### DISCUSSION TITLE
Unbearable lag when working with a vector image

### DISCUSSION THREAD

**Discussion by Ectophasm** [keiwando/toonsquid#428] [General]
I'm using a new 11 inch iPad Pro, totally up to date with the app and operating system. There is lag when using the vector brush that causes inaccurate strokes, and also lag when moving / shifting view of the canvas.

I initially started with a totally zoomed in canvas using a 1 px black vector brush. I enlarged the image and used the eraser in combination with an off-white color to sharpen up the image, then shrunk it back down. Is my only option to not work with vector art this way, or is there a solution?

**Comment by Cat-Butler** [comment]
Just out of curiosity, what size is your canvas?
I'm just guessing, but I suspect that ToonSquid is optimised more for animation-sized canvases, rather than print sizes.

**Reply by Ectophasm** [reply]
Edit: It's not canvas size. I tried putting it into a 400x400 with the same issue.

**Reply by keiwando** [reply]
The scene resolution won't have any impact on the rendering performance of vector layers in the editor. All vector layer contents are rendered at the display resolution inside of the editor.

**Comment by keiwando** [comment]
Without seeing the specific project and the layer structure and contents it's nearly impossible for me to say what exactly is causing the slowdowns, so if you have the project and still care to know, feel free to send a copy of it to contact@toonsquid.com and I can take a closer look and provide more info.
Vector layers definitely get more CPU-intensive to render the more brush strokes you have and also the more curves each of the brush stroke shapes is made up of. That might be the main issue here. There are also various optimizations for rendering vector layers that cannot be performed under certain circumstances. When working with vector layers a lot, I would generally recommend disabling onion skinning on layers for which you don't need it and keeping the number of onion skin frames to a minimum.

**Reply by Ectophasm** [reply]
You are right, I was going overboard on brushstrokes and shouldn't have been using vector art over raster in the first place for my purposes.
